modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/ExtensionsClientTestCase.java

```diff
@@ -0,0 +1,265 @@
+    assertThat(result.getOutput(), instanceOf(List.class));
+    assertThat(result.getOutput(), hasSize(0));
+  }
+
+  @Test
+  @Description("Executes an operation that has a @NullSafe annotated parameter using the client and checks the output")
+  public void executeOperationWithNullSafeParameter() throws MuleException {
+    OperationParameters params = builder().configName("banana").build();
+    Result<VeganPolicy, Attributes> result = client.execute(VEGAN, "applyPolicy", params);
+    assertThat(result.getOutput().getMeetAllowed(), is(false));
+    assertThat(result.getOutput().getIngredients().getSaltMiligrams(), is(0));
+  }
+
+  @Test
+  @Description("Executes an operation that returns attributes metadata using the client and checks the output and the attributes")
+  public void executeOperationThatReturnsAttributes() throws MuleException {
+    OperationParameters params = builder().configName(HEISENBERG_CONFIG).build();
+    Result<String, IntegerAttributes> result = client.execute(HEISENBERG_EXT_NAME, "getEnemy", params);
+    assertThat(result.getOutput(), is("Gustavo Fring"));
+    assertThat(result.getAttributes().isPresent(), is(true));
+    assertThat(result.getAttributes().get().getValue(), is(0));
+  }
+
+  @Test
+  @Description("Executes a void operation using the client")
+  public void executeVoidOperation() throws MuleException {
+    OperationParameters params = builder().configName(HEISENBERG_CONFIG).build();
+    Result<Object, Attributes> result = client.execute(HEISENBERG_EXT_NAME, "die", params);
+    assertThat(result.getOutput(), is(nullValue()));
+    assertThat(result.getAttributes().isPresent(), is(true));
+    assertThat(result.getAttributes().get(), instanceOf(NullAttributes.class));
+  }
+
+  @Test
+  @Description("Executes an operation that fails using the client and checks the throwed exception")
+  public void executeFailureOperation() throws MuleException {
+    exception.expectCause(instanceOf(ConnectionException.class));
+    exception.expectMessage("You are not allowed to speak with gus.");
+    OperationParameters params = builder().configName(HEISENBERG_CONFIG).build();
+    client.execute(HEISENBERG_EXT_NAME, "callGusFring", params);
+  }
+
+  @Test
+  @Description("Executes an operation asynchronously using the client and checks the output")
+  public void executeOperationWithNonBlocking() throws MuleException {
+    Reference<Boolean> success = new Reference<>();
+    success.set(false);
+    OperationParameters params = builder().configName(HEISENBERG_CONFIG)
+        .addParameter("greeting", "jeje")
+        .addParameter("age", 23)
+        .addParameter("name", "Juani")
+        .addParameter("knownAddresses", emptyList())
+        .build();
+
+    Publisher<Result<String, Attributes>> publisher = client.executeAsync(HEISENBERG_EXT_NAME, "alias", params);
+    publisher.subscribe(new Subscriber<Result<String, Attributes>>() {
```

Comment:
No need to create a new subscriber here.  Instead use `Mono.from(publisher).doOnNext(..).doOnError(..).subscribe();`.  We are avoiding using Reactor in API, but fine to use in implementation/tests.